icechunk/src/storage/mod.rs

```diff
@@ -223,17 +238,44 @@ where
     s.try_filter_map(|info| async move { Ok(convert_list_item(info)) }).boxed()
 }
 
+/// Split an object request into multiple byte range requests
+///
+/// Returns tuples of (offset, size) for each request. It tries to generate the maximum number of
+/// requests possible, not generating more than `max_parts` requests, and each request not being
+/// smaller than `min_part_size`.
```

Comment:
```suggestion
/// smaller than `min_part_size`. Note that the size of the last request is >= the preceding one.
```

(i assumed the other way, until I read the property test)

Reply:
done